gamemodes/terrortown/gamemode/client/cl_main.lua

```diff
@@ -729,16 +731,15 @@ net.Receive("TTT2PlayerAuthedShared", function(len)
 	local steamid64 = net.ReadString()
 	local name = net.ReadString()
 
+	-- checking for bots
+	if steamid64 == "" then
+		steamid64 = nil
+	end
+
 	hook.Run("TTT2PlayerAuthed", steamid64, name)
 end)
 
 hook.Add("TTT2PlayerAuthed", "TTT2CacheAvatar", function(steamid64, name)
-	local ply = player.GetBySteamID64(steamid64)
-
-	if not IsValid(ply) or ply:IsBot() then
-		steamid64 = nil
-	end
-
 	draw.CacheAvatar(steamid64, "medium") -- caching
```

Comment:
Also, is this still needed? Or rather is this whole function even needed?

Reply:
All avatars should be fetched before they are first used.

When a client joins it caches the avatars of all existing players.
To this client any new joining players would be avatarless though and that's why these are cached.

I should probably also add 
```
draw.CacheAvatar(steamid64, "small")
draw.CacheAvatar(steamid64, "large")
```